### ISSUE TITLE
Split to parts slicing issue

### ISSUE THREAD

**Issue by schurter-richard** [bambulab/BambuStudio#4241]
### Bambu Studio Version

1.9.1.66

### Where is the application from?

Bambu Lab Official website

### OS version

Windows 11

### Additional system information

Lenovo Thinkpad X1 Yoga Gen 7

### Printer

X1C

### How to reproduce

-Have a part with two pieces that need to be printed, in my case it is a single layer of a different color in the middle of a print.
-After importing to Bambu Studio, split to parts. Then make the single layer a different filament.
-Slice the part
-Check the layer that is a different color

### Actual results

Instead of getting a single uniform layer, it takes on the pattern of the layers above/below it. This causes overhangs to fail, and the print to be ugly. One way to get around this if the first layer is the one with a different color is to split to object instead, but this does not work if the second color is not on the first layer.
<img width="629" alt="Sliced as Parts Close Up" src="https://github.com/bambulab/BambuStudio/assets/112225708/bec8cc8a-ec1e-4dc5-acfd-43e226df41f0">
<img width="643" alt="Sliced as Parts" src="https://github.com/bambulab/BambuStudio/assets/112225708/fd3a6d9f-57b4-4ae5-9a50-4d2a09ed7764">

 <img width="646" alt="Bambulab Split to Parts Issue" src="https://github.com/bambulab/BambuStudio/assets/112225708/a1e5e9b4-7695-4529-adc0-5c03f2892249">

### Expected results

I would expect the layer to be printed as if it were the bottom/top layer. If you look at the image where I split it to objects it works great. The problem is I need the secondary color to be in the middle of the part, which does not work when splitting to object.
<img width="647" alt="Sliced as Objects" src="https://github.com/bambulab/BambuStudio/assets/112225708/5db23e6b-4016-4dfb-b3f5-7f8e0ed60155">

### Project file & Debug log uploads

[debug_Sat_Jun_08_13_52_57_35660.log.zip](https://github.com/user-attachments/files/15749926/debug_Sat_Jun_08_13_52_57_35660.log.zip)
[Bambu Studio Split to part slicing issue.zip](https://github.com/user-attachments/files/15749929/Bambu.Studio.Split.to.part.slicing.issue.zip)

### Checklist of files to include

- [X] Log file
- [X] Project file

**Comment by zhimin-zeng-bambulab** [comment]
![image](https://github.com/bambulab/BambuStudio/assets/120697589/cec1928f-0cd2-43e9-9aa5-059a12defca1)

@schurter-richard  Does your problem mean that the print quality in these areas is poor?

**Comment by schurter-richard** [comment]
> ![image](https://private-user-images.githubusercontent.com/120697589/339191562-cec1928f-0cd2-43e9-9aa5-059a12defca1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgyNDc5MjgsIm5iZiI6MTcxODI0NzYyOCwicGF0aCI6Ii8xMjA2OTc1ODkvMzM5MTkxNTYyLWNlYzE5MjhmLTBjZDItNDNlOS05YWE1LTA1OWExMmRlZmNhMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYxM1QwMzAwMjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01ODRlNDNjOGY5ODFhY2FlNDZmMjY0MDVlMWFiMzQ2NWU3YjIzYzZmNWQ2MWYyN2RjY2NjMTdlY2Q3ODUxZjFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ZtZiK6ZOeCLPdqR3lB9rd29Yqt4ibDP2-aufv9OPQXA)
> 
> @schurter-richard Does your problem mean that the print quality in these areas is poor?

Yes, poor quality like shown in your image. And it also will leave gaps between extrusion lines depending on the model being printed.
